### PR TITLE
OU-257: Migrate Developer View > Observe > Silences Tab from openshift/console to openshift/monitoring-plugin

### PR DESCRIPTION
**Pull request by zhuje**
### OU JIRA 
https://issues.redhat.com/browse/OU-257

### CONSOLE JIRA 
https://issues.redhat.com/browse/CONSOLE-4330
This duplicate issue was created because openshift/console github bots require a valid CONSOLE Jira to be associated with all PRs. 

### Corresponding PR to remove 'Silences' tab from `openshift/console`
https://github.com/openshift/console/pull/14480

## Description 
To consolidate the code of the` Developer View > Observe > Silences Tab` is being moved from `openshift/console` to `openshift/monitoring-plugin`.

- This PR renders the 'Silences' tab in the openshift/monitoring-plugin. 
- The corresponding [PR](https://github.com/openshift/console/pull/14480) removes the 'Silences' tab in the openshift/console. 

## Testing Notes
- This PR should be tested with https://github.com/openshift/console/pull/14480
- All functionalities should be the same as before. No new features were added.

### CODE REVIEW

**Comment by zhuje** [comment]
/hold 

PENDING QA after code review

**Review by PeterYurkovich — CHANGES_REQUESTED** [review]
Apart of our discussion on the migration stories (especially for the alerting parts) was to refactor the alerting pages to be split up more as the alerting.tsx file exceeded 1400 loc. Could you please pull the silences pages listed [here](https://github.com/openshift/monitoring-plugin/blob/main/web/src/components/alerting/README.md) out of the alerting.tsx and into separate files?

File: web/src/components/utils.ts [hunk 1]
```diff
@@ -119,15 +119,20 @@ export const refreshSilences = (
...
   perspective: Perspective,
   silencesKey: silencesKey,
+  namespace?: string,
 ): void => {
   const { alertManagerBaseURL } = window.SERVER_FLAGS;
   if (!alertManagerBaseURL) {
     return;
   }
 
+  if (perspective == 'dev' && !namespace) {
```
Comment (PeterYurkovich) [inline web/src/components/utils.ts:129]: NIT: `perspective === 'dev'`

**Review by PeterYurkovich — APPROVED** [review]

File: web/src/components/utils.ts (hunk 1 above)
Comment (zhuje) [reply web/src/components/utils.ts]: Updated in latest commit

**Comment by zhuje** [comment]
I pushed again to fix a bug. The SilencesPage wasn't rendering under the 'Silences' Tab. I updated the `console-package.json` to fix this. 

image

File: web/console-extensions.json [hunk 2]
```diff
@@ -109,6 +109,33 @@
...
     }
   },
+  {
+    "type": "console.tab",
+    "properties": {
+      "contextId": "dev-console-observe",
+      "name": "%plugin__monitoring-plugin~Silences%",
+      "href": "silences",
+      "component": {
+        "$codeRef": "SilencesPage"
```
Comment (PeterYurkovich) [inline web/console-extensions.json:119]: Is there a reason this code ref needs to be different from the MonitoringUI entrypoint that the rest use?

Comment (zhuje) [reply web/console-extensions.json]: I updated this to use the MontioringUI entry point that the rest use in the latest push.

**Comment by zhuje** [comment]
Requesting review from QA contact:
/cc @juzhao

**Comment by juzhao** [comment]
tested
 https://github.com/openshift/monitoring-plugin/pull/252
 https://github.com/openshift/console/pull/14480
 with cluster-bot
`launch 4.18.0-0.nightly-2024-11-14-090045,openshift/monitoring-plugin#252,openshift/console#14480 aws`
there is one regression bug: [OCPBUGS-44594](https://issues.redhat.com/browse/OCPBUGS-44594)

**Comment by zhuje** [comment]
@juzhao I pushed some new changes. Could you test again? 

I think the issue was that the DELETE request didn't include the namespace initially (e.g. `namespace=undefined` from DELETE `/api/alertmanager-tenancy/api/v2/silence/1f5d9fdc-4d07-4093-8119-c4f3f8c138e1?namespace=undefined`). 

I tested using ClusterBot `launch 4.18,openshift/monitoring-plugin#252,openshift/console#14480 aws` and took a short video of new changes: https://drive.google.com/drive/folders/1Nu-4snd4EiyZ3PPIZHUNZ6lcP8mhQJvI. 

The plugin should now request DELETE `/api/alertmanager-tenancy/api/v2/silence/1f5d9fdc-4d07-4093-8119-c4f3f8c138e1?namespace=<namespace>` (e.g. DELETE `/api/alertmanager-tenancy/api/v2/silence/1f5d9fdc-4d07-4093-8119-c4f3f8c138e1?namespace=monitoring-plugin`).

**Comment by juzhao** [comment]
tested again
`launch 4.18,openshift/monitoring-plugin#252,openshift/console#14480 aws`
[OCPBUGS-44594](https://issues.redhat.com/browse/OCPBUGS-44594) is fixed

**Pull request merged**